src/components/views/messages/RoomCreate.tsx

```diff
@@ -37,31 +40,57 @@ interface IProps {
  * room.
  */
 export const RoomCreate: React.FC<IProps> = ({ mxEvent, timestamp }) => {
+    // Note: we ask the room for its predecessor here, instead of directly using
+    // the information inside mxEvent. This allows us the flexibility later to
+    // use a different predecessor (e.g. through MSC3946) and still display it
+    // in the timeline location of the create event.
+    const roomContext = useContext(RoomContext);
+    const predecessor = useRoomState(
+        roomContext.room,
+        useCallback((state) => state.findPredecessor(), []),
+    );
+
     const onLinkClicked = useCallback(
         (e: React.MouseEvent): void => {
             e.preventDefault();
 
-            const predecessor = mxEvent.getContent()["predecessor"];
-
             dis.dispatch<ViewRoomPayload>({
                 action: Action.ViewRoom,
-                event_id: predecessor["event_id"],
+                event_id: predecessor.eventId,
                 highlighted: true,
-                room_id: predecessor["room_id"],
+                room_id: predecessor.roomId,
                 metricsTrigger: "Predecessor",
                 metricsViaKeyboard: e.type !== "click",
             });
         },
-        [mxEvent],
+        [predecessor?.eventId, predecessor?.roomId],
     );
-    const predecessor = mxEvent.getContent()["predecessor"];
-    if (predecessor === undefined) {
-        return <div />; // We should never have been instantiated in this case
+
+    if (!roomContext.room || roomContext.room.roomId !== mxEvent.getRoomId()) {
+        logger.warn(
+            "RoomCreate unexpectedly used outside of the context of the room containing this m.room.create event.",
+        );
+        return <></>;
     }
-    const prevRoom = MatrixClientPeg.get().getRoom(predecessor["room_id"]);
-    const permalinkCreator = new RoomPermalinkCreator(prevRoom, predecessor["room_id"]);
+
+    if (!predecessor) {
+        logger.warn("RoomCreate unexpectedly used in a room with no predecessor.");
+        return <div />;
```

Comment:
Is that unexpected? If so can we rename it to RoomCreateWithPredecessor or something equally obvious

Reply:
Yes, it is unexpected. I'd like to do the rename separately to keep this change clean.